cmds/live.c

```diff
@@ -238,6 +238,69 @@ static int query_agent_capabilities(int fd)
 	return status;
 }
 
+/**
+ * forward_option - forward a given option and its data to the agent
+ * @fd           - agent socket file descriptor
+ * @capabilities - features supported by the agent
+ * @opt          - option type to send
+ * @value        - option data load
+ * @value_size   - size of @value
+ * @return       - status: -1 on error, >= 0 on success
+ */
+static int forward_option(int fd, int capabilities, int opt, void *value, size_t value_size)
+{
+	void *data;
+	size_t data_size;
+	struct uftrace_msg ack;
+	int ret = -1;
+
+	if (opt && capabilities == 0) {
```

Comment:
I don't know what you want to check exactly.  How do you change it if we have non-zero capabilities?

Reply:
Sorry, that is meant to be a bitwise AND.

I want to check if the option that the client is forwarding is supported by the agent.

`capabilities` stores the supported agent features as OR-ed flags, e.g. `capabilities = OPT_TRACE | OPT_FILTER | OPT_DEPTH`.

The above example means that the agent is able to change the tracing state (on/off), to apply filters, and to change the filter depth.